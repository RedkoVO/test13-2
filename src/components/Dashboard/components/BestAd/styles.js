export default theme => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    marginTop: '40px',
    width: '100%',

    [theme.breakpoints.down('md')]: {
      alignItems: 'flex-start',
      justifyContent: 'center',
      flexWrap: 'wrap',
      marginTop: 0
    }
  },
  bestItem: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    paddingBottom: '40px',

    [theme.breakpoints.down('md')]: {
      marginRight: '20px',
      marginBottom: '20px'
    }
  },
  titleWrap: {
    width: '100%',
    paddingLeft: '40px',
    paddingBottom: '10px',
    boxSizing: 'border-box',

    [theme.breakpoints.down('md')]: {
      display: 'flex',
      justifyContent: 'center',
      paddingLeft: 0
    }
  },
  bestTitle: {
    fontSize: '37px',
    fontWeight: 700,
    color: '#5e7cff',
    textAlign: 'center',
    marginBottom: '0px',
    textDecoration: 'none',
    margin: '0 auto'
  },
  bestContent: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  bestImg: {
    display: 'block',
    borderRadius: '15px',
    width: '335px',
    height: '225px',
    boxShadow: '0px 10px 20px 0px rgba(0,0,0,0.2)',
    marginBottom: '25px'
  },
  bestItemImg: {
    width: '100%',
    height: '100%',
    borderRadius: '15px'
  },
  bestLink: {
    fontSize: '22px',
    fontWeight: 700,
    color: '#575757',
    textAlign: 'center',
    marginBottom: '0px',
    textDecoration: 'none',
    margin: '0 auto',

    '& span': {
      fontSize: '37px'
    }
  }
})
