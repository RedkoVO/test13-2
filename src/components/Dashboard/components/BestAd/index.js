import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

const BestAd = ({ classes, bestAds }) => {
  return (
    <section id="shops">
      <div className={classes.bestTitle}>Shops</div>
      <div className={classes.root}>
        {bestAds.map(item => (
          <div className={classes.bestItem} key={item.id}>
            <div className={classes.titleWrap}>
              <Link to={item.link} className={classes.bestTitle}>
                {item.title}
              </Link>
            </div>
            <div className={classes.bestContent}>
              <Link to={item.link} className={classes.bestImg}>
                <img src={item.img} alt="" className={classes.bestItemImg} />
              </Link>
              <Link to={item.link} className={classes.bestLink}>
                {item.titleLink}: <br />
                <span>{item.nameLink}</span>
              </Link>
            </div>
          </div>
        ))}
      </div>
    </section>
  )
}

BestAd.propTypes = {
  classNamees: PropTypes.object,
  bestAds: PropTypes.array
}

export default withStyles(styles)(BestAd)
