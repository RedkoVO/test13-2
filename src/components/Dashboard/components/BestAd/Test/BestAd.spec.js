import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import BestAd from '../'

import BestGameLogo from '../../../../../assets/images/best-game.png'

const data = [
  {
    id: 1,
    title: 'Games',
    img: BestGameLogo,
    link: '/',
    titleLink: 'Game of the day',
    nameLink: 'Smash Champs'
  },
  {
    id: 2,
    title: 'Shop',
    img: BestGameLogo,
    link: '/',
    titleLink: 'Product of the day',
    nameLink: 'Blue cool Jelly'
  }
]

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <BestAd bestAds={data} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
