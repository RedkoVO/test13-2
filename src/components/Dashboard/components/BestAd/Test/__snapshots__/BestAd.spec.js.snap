// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="shops"
>
  <div
    className="BestAd-bestTitle-4"
  >
    Shops
  </div>
  <div
    className="BestAd-root-1"
  >
    <div
      className="BestAd-bestItem-2"
    >
      <div
        className="BestAd-titleWrap-3"
      >
        <a
          className="BestAd-bestTitle-4"
          href="/"
          onClick={[Function]}
        >
          Games
        </a>
      </div>
      <div
        className="BestAd-bestContent-5"
      >
        <a
          className="BestAd-bestImg-6"
          href="/"
          onClick={[Function]}
        >
          <img
            alt=""
            className="BestAd-bestItemImg-7"
            src="best-game.png"
          />
        </a>
        <a
          className="BestAd-bestLink-8"
          href="/"
          onClick={[Function]}
        >
          Game of the day
          : 
          <br />
          <span>
            Smash Champs
          </span>
        </a>
      </div>
    </div>
    <div
      className="BestAd-bestItem-2"
    >
      <div
        className="BestAd-titleWrap-3"
      >
        <a
          className="BestAd-bestTitle-4"
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </div>
      <div
        className="BestAd-bestContent-5"
      >
        <a
          className="BestAd-bestImg-6"
          href="/"
          onClick={[Function]}
        >
          <img
            alt=""
            className="BestAd-bestItemImg-7"
            src="best-game.png"
          />
        </a>
        <a
          className="BestAd-bestLink-8"
          href="/"
          onClick={[Function]}
        >
          Product of the day
          : 
          <br />
          <span>
            Blue cool Jelly
          </span>
        </a>
      </div>
    </div>
  </div>
</section>
`;
